### DISCUSSION TITLE
Page types: Do we need a "catch-all" page type?

### DISCUSSION THREAD

**Discussion by Elchi3** [mdn/mdn-community#322] [Content]
This came up in our London work week. Right now the sort of fallback page-type is "guide" if nothing else fits. Is this ideal? Should we introduce a dedicated "catch-all" page-type for this purpose?

Link to the meeting minutes: https://github.com/openwebdocs/project/blob/main/steering-committee/meetups/london-2023/session-ia.md#catchall-page-type

**Comment by bsmth** [comment]
I like Will's suggestion that all pages should have a `page-type` as this means content is added with intention. Falling back to "guide" everywhere might indicate there's a missing page type.

Pages with "guide" where the content is not really a guide would indicate that it's either flagged for deletion or it should be another page type.

Are there some examples we could see? Is it mostly pages that are solely collections of links to other resources?

**Reply by wbamberg** [reply]
> I like Will's suggestion that all pages should have a `page-type` as this means content is added with intention. Falling back to "guide" everywhere might indicate there's a missing page type.
> 
> Pages with "guide" where the content is not really a guide would indicate that it's either flagged for deletion or it should be another page type.
> 
> Are there some examples we could see? Is it mostly pages that are solely collections of links to other resources?

No, these have a `landing-page` type.

There are a few cases of pages which could arguably have a type but where I just felt the need wasn't that great.

For example, CSS has "recipes", like https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Media_objects . I was a bit uncertain whether to define a page type for that. They do have a specific structure.

Another example would be in Learn, where we could have a "learn-module" type for pages like https://developer.mozilla.org/en-US/docs/Learn/JavaScript/First_steps and a "learn-article" (or "learn-lesson" or something) type for pages like https://developer.mozilla.org/en-US/docs/Learn/JavaScript/First_steps/A_first_splash ), and a "learn-assessment" type for pages like https://developer.mozilla.org/en-US/docs/Learn/JavaScript/First_steps/Silly_story_generator. Again, these pages have a structure.

I think it is a matter of judgement in cases like this whether to define dedicated page types or not. The argument against is that the more page types, the more complicated the platform gets. If we have hundreds of types then the complexity is just going to be overwhelming. In the case of CSS recipes I chose not to (after flagging the question in the related issue) with the idea that we could easily break them out later if it seemed worthwhile. We haven't done Learn yet (but the case for page types seems stronger there).

As it stands, having added page types to probably 9000 pages, I haven't seen cases where I am really uncomfortable with any of the `page-type: guide` assignments.

It's not exactly that "guide" is a catch-all, perhaps that was an unfortunate term. It's more that "guide" doesn't prescribe much in terms of structure or semantics, compared with the other types. So it's easy for a page to be a guide page, if it isn't one of the other types.

I think you're right, Brian, that if a page looks unhappy with `page-type: guide` then that probably means we should consider a new type.
